app/javascript/packs/puerto-rico-guidance.ts

```diff
@@ -13,5 +17,30 @@ function onStateSelectionChange() {
   showOrHidePuertoRicoExtras(stateSelector?.value);
 }
 
+function onIdentityDocStateSelection() {
+  const stateSelectors = document.querySelectorAll<HTMLSelectElement>('.usa-select');
+  stateSelectors.forEach((stateSelector) => {
+    if (!stateSelector.id.includes('jurisdiction')) {
+      stateSelector.addEventListener('change', () =>
+        showOrHidePuertoRicoExtras(stateSelector.value),
+      );
+      showOrHidePuertoRicoExtras(stateSelector.value);
+    }
+  });
+}
+
+function onUpdate() {
+  const updateStateSelectors = document.querySelectorAll<HTMLSelectElement>('.usa-select');
+  updateStateSelectors.forEach((updatedStateSelector) => {
+    if (!updatedStateSelector.id.includes('jurisdiction')) {
+      showOrHidePuertoRicoExtras(updatedStateSelector.value);
+    }
+  });
+}
+
 document.getElementById('idv_form_state')?.addEventListener('change', onStateSelectionChange);
-onStateSelectionChange(); // set initial visibility
+
+// set initial visibility
+onStateSelectionChange();
+onUpdate();
+onIdentityDocStateSelection();
```

Comment:
It would be good to ensure this is deferred until at least after [`DOMContentLoaded`](https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event), though based on the way JS is currently handled it may be alright.